hyrax/app/assets/javascripts/hyrax/save_work/required_fields.es6

```diff
@@ -7,80 +11,68 @@ export class RequiredFields {
+    var missing = this.requiredFields.filter((n, elem) => { return this.isRequiredButBlank(elem) } )
+    return missing.length === 0
 	}
 	
-	requiredIf() {
-		// this is hard-coded - to do, use erb to grab a list
-		let conditions = ['any', 'name']
-		var elems = []
-
-		// build the list of elem; cleanup
-		$.each(conditions, function(n,value) {
-			$(":input.require-if-"+value).each(function(n,elem) {
-				// Remove the required class, attribute and box-shadow (they will be reset if needed)
-				$(elem).removeClass('required');
-				$(elem).attr('required', null);
-				$(elem).removeAttr('style');
-				elems.push({ value : elem })
-			});
-		});
+  // helper to clear the 'required' styling
+  notRequired(elem, label) {
+    $(label).removeClass('required')
+  	$(label).find('span').remove()
 		
-		$.each(elems, function(n,elem_obj) {
-				var required = elem_obj.value
-				var id = $(required).attr('id');
-				var parents = $(required).closest('div.multi-nested')
+  	$(elem).removeClass('required');
+	  $(elem).attr('required', null);
+	  $(elem).removeAttr('style');
+  }
 
-				if (elem_obj.key === 'any') {
-					var label = $(parents).find('label[for="' + id + '_"]')
-				} else {
-					var label = $(parents).find('label[for="' + id + '_' + elem_obj.key + '"]')
-				}
+  required(elem, label) {
+    $(label).addClass('required control-label')
+		$(label).append('<span class="label label-info required-tag">required</span>')
+		$(elem).addClass('required');
+		$(elem).attr('required', 'required');
+		$(elem).css('box-shadow','0px 0px 0px 2px red');
+  }
 
-				// Remove the required class and span (they will be reset if needed)
-				$(label).removeClass('required')
-				$(label).find('span').remove();
-				
-				// Find the immediate field-wrapper
-				// Gather the inputs
-				this.inputs = $(required).closest('li.field-wrapper').find(':input');
-				// Skip the current elem, any hidden fields and any buttons, skip operator (role), Processed (date) (pre-filled but should not be required)
-				if (this.inputs.filter((n, el) => {
-					if ( 
-						!el.parentNode.className.includes("hidden") && !el.className.includes('btn') && !el.className.includes('remove-hidden') && el != required && el.value !== 'operator' && el.value !== 'Processed' ) {
-						return (el.value !== '');
-					}
-				} ).length > 0) {
-					// Add the required classes, attributes and labels
-					if ( $(label).has('span').length === 0 ) {
-						$(label).addClass('required control-label')
-						$(label).append('<span class="label label-info required-tag">required</span>');
-						$(required).addClass('required');
-						$(required).attr('required', 'required');
-						// Only add the css if the value is empty
-						if ( ($(required).val() === null) || ($(required).val().length < 1) ) {
-							$(required).css('box-shadow','0px 0px 0px 2px red');
-						}
-					}
-				};
-		});
+	isRequiredButBlank(elem) {
```

Comment:
renamed the hyrax function isValuePresent because it kept confusing me (boolean names are hard). this function still does what it used to do (if a field is simply required w/o the special data attribute and it is not filled in, then the checkmark on the right bar stays red)

it now also checks if the data-required attribute is present.  If it is then it does the required if functionality